config/configmap/inferenceservice.yaml

```diff
@@ -143,14 +143,24 @@ data:
         "cpuLimit": "1",
         "storageSpecSecretName": "storage-config"
     }
+  # For a quick reference about AWS ENV variables:
+  # AWS Cli: https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html
+  # Boto: https://boto3.amazonaws.com/v1/documentation/api/latest/guide/configuration.html#using-environment-variables
   credentials: |-
     {
        "gcs": {
            "gcsCredentialFileName": "gcloud-application-credentials.json"
        },
        "s3": {
            "s3AccessKeyIDName": "AWS_ACCESS_KEY_ID",
```

Comment:
few questions:
1. there are 2 variables used to construct the service endpoint URL: `s3Endpoint` and `s3UseHttps`. Depending on the value of useHttps the final AWS_ENDPOINT_URL is set. @yuzisun Is there a particular reason to do it this way? Just for my information
2. how is s3VerifySSL used?
3. In _Test B: Deploy S3 Example using only service account (no secret) configuring all values inside inferenceservice configmap_, does the user still need to keep "s3AccessKeyIDName": "AWS_ACCESS_KEY_ID" and "s3SecretAccessKeyName": "AWS_SECRET_ACCESS_KEY", in configmap? It can be a little confusing

Reply:
for 1) I think this is mostly for backwards compatibility and integration with Kubeflow environment, for example [tfserving was using S3_ENDPOINT](https://github.com/kubeflow/examples/blob/master/code_search/kubeflow/vendor/kubeflow/tf-serving@dddba26893846041b7cee25ee6121e02a04bd503/prototypes/tf-serving-aws.jsonnet#L51), I am not sure if this is still the case currently.
for 2) Same as other setting, when `s3VerifySSL` is set to false it then does not verify the server side certifications which is mostly for testing purpose.
for 3) I do not think these are needed if we are using service account without secret, @matty-rose maybe we can add a comment in the configmap to state which settings are only needed for static credentials?

Reply:
yep for 3. those fields are not required in the configmap if not using static credentials, have added a comment to make this clearer

Reply:
1. One change I would recommend based on #2377 is to remove the default value for all the fields except access and secret key name, especially s3Endpoint. This PR lgtm once this change is made.
1. Thanks for adding the comment in this file. I think this should go to the website PR as well and also add information about default values used by each or atleast some of these like https, endpoint, region in website. I will try to review that one today.
1. I could not find any reference for `S3_VERIFY_SSL` variable in boto3 or aws go sdk docs. So, I am not sure how its being used. Dont want to block the PR for this though

Reply:
I have made the change removing the endpoint/use https default values and will try find some time this week to make further updates to the website PR to describe the use of these better. Thanks for your review @surajkota

Reply:
@surajkota `S3_VERIFY_SSL`/`S3_ENDPOINT`/`S3_USE_HTTPS` were originally inherited from tensorflow/tfserving(see https://docs.w3cub.com/tensorflow~guide/deploy/s3). I think we may no longer need these as we always use storage initializer to download models to a local volume and tfserving runtime never directly use these environment variables to get models from s3 in kserve. I created an [issue](https://github.com/kserve/kserve/issues/2397) to track this to clean up these variables.

Reply:
Ah, got it now

Reply:
Approved the PR, maybe we should remove `s3VerifySSL` from this configmap since it will not be used or talk abou it on the website for now